astropy/units/quantity.py

```diff
@@ -902,6 +902,52 @@ def to_value(self, unit=None, equivalencies=[]):
     to_value : Get the numerical value in a given unit.
     """)
 
+    def to_string(self,unit='human'):
```

Comment:
As written, this is not correct, as it should return a string. There is also no reason to duplicate the whole logic from `utils/console` - I think there should be just one way to do it. Finally, the default should be `None`, i.e., do not do a unit transformation, but just do what is now done in `__str__` (which could thus call `self.to_string()`). I'm thinking here in analogy with how this is done in `Angle`, i.e., we should also have a `format` which can be `latex`, etc.

Aside: I'm a bit worried we're now making the PR complicated because we're mixing having a nice utility that converts to a handy unit with defining a new method on `Quantity` for making a string. It may make sense to keep this one to just the utility method, and then do `to_string()` separately.

Reply:
@mhvk you are right there no need for duplicating the logic. I will correct it along with all the changes suggested . By the way I din't duplicate the logic i just made changes in the new location without removing the code from previous location (a foolish thing to do).

Reply:
No worries. As said, I think it makes most sense to first implement the helper function, and then think about how best to make it visible in a new `Quantity.to_string()` method.